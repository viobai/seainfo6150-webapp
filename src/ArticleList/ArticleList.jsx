import React from "react";
import PropTypes from "prop-types";
import ArticleListItem from "../ArticleListItem/ArticleListItem";

const ArticleList = (props) => {
    let articlesList;

    if (props.articles.length) {
        articlesList = (
        <ul>
          {props.articles.map(element => (
            <li key={element.slug}>
              <ArticleListItem article={element}/>
            </li>
          ))}
        </ul>
      );    
    } else {
        articlesList = <p>You have no data!</p>
    }

    return (
      <div>
          {articlesList}
      </div>
    );
    
  };

  ArticleList.propTypes = {
    articles: PropTypes.array.isRequired
  };

export default ArticleList;