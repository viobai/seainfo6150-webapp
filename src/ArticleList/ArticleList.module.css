.listContainer {
    background: rgb(155, 155, 155);
    border: 1px dashed rgb(100,100,100);
    border-radius: 10px;
    box-shadow: 0px 0px 10px 0px red;
    margin: 30px 2px;
}

ul {
    list-style-type: none
}
  
@supports (display: grid) {
    ul {
        display: grid;
        grid-template-columns: repeat(3, 1fr);
        gap: 20px;
    }
}