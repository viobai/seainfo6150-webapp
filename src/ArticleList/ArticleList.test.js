import React from "react";
import ArticleList from "./ArticleList";

describe("ArticleList tests", () => {
    it("renders correctly with two articles", () => {
        var articles = [
            {
                "title": "The Statue of Liberty's torch heads to new museum",
                "slug": "181122102925-statue-of-liberty-crown",
                "shortText": "The Statue of Liberty's original torch is getting a new home",
                "timeStamp": "2018-11-22T15:12:24.000Z",
                "displayDate": "November 22nd 2018, 7:12 am",
                "image": {
                    "_url": "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
                }
            },
            {
                "title": "You can take a pie on a plane",
                "slug": "170321135754-airport-security-line",
                "shortText": "Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?",
                "timeStamp": "2018-11-21T22:57:55.000Z",
                "displayDate": "November 21st 2018, 2:57 pm",
                "image": {
                    "_url": "https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
                }
            }
        ];
        const { container } = render(<ArticleList articles={articles}/>);
        expect(container).toMatchSnapshot();
    });
});