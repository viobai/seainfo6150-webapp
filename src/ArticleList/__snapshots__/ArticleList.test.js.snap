// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly with two articles 1`] = `
<div>
  <div
    class="listContainer"
  >
    <ul>
      <li
        class="item"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <img
              alt="The Statue of Liberty's torch heads to new museum"
              class="img"
              src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
            />
            <a
              class="title"
              href="/articlelist/181122102925-statue-of-liberty-crown"
            >
              <h2
                class="title"
              >
                The Statue of Liberty's torch heads to new museum
              </h2>
            </a>
          </div>
          <button
            class="button"
          >
            Show More
          </button>
        </article>
      </li>
      <li
        class="item"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <img
              alt="You can take a pie on a plane"
              class="img"
              src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
            />
            <a
              class="title"
              href="/articlelist/170321135754-airport-security-line"
            >
              <h2
                class="title"
              >
                You can take a pie on a plane
              </h2>
            </a>
          </div>
          <button
            class="button"
          >
            Show More
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
